prow/plugins/plugins.go

```diff
@@ -363,10 +363,16 @@ type Cat struct {
 	KeyPath string `json:"key_path,omitempty"`
 }
 
+// Label is the config for the label plugin.
 type Label struct {
 	// AdditionalLabels is a set of additional labels enabled for use
 	// on top of the existing "kind/*", "priority/*", and "area/*" labels.
-	AdditionalLabels []string `json:"additional_labels"`
+	// Labels can be used with `/[remove-]label <additionalLabel>` commands.
+	AdditionalLabels []string `json:"additional_labels,omitempty"`
+	// Prefixes is a set of label prefixes which is added to the list
+	// of existing label prefixes like "kind", "priority" and "area".
+	// Labels can be used with `/[remove-]<prefix> <target>` commands.
+	Prefixes []string `json:"prefixes,omitempty"`
```

Comment:
Should this list be additive to `area|committee|kind|priority|sig|triage|wg` or be authoritative?

If the latter, we can set `area|committee|kind|priority|sig|triage|wg` as the defaults in `setDefaults()`.

Reply:
Also, if it's authoritative, we can construct another regex for custom prefix based labels during config load rather than in `handle()`...which seems cleaner to me.

Reply:
Or even if it's additive, we can do it during config load. I should probably brainstorm before sending the PR. :)

Reply:
> Also, if it's authoritative, we can construct another regex for custom prefix based labels during config load rather than in handle()...which seems cleaner to me.

Err, I am not sure if that's really cleaner. This would mean adding two fields in the `Label` config. Something like `LabelRe` and `RemoveLabelRe` which are of type `*regexp.Regexp`, which seems strange because we don't expose any regexp to the user.

Leave the commit as-is for now. Will iterate based on reviews.

Reply:
So thinking about this as from a "prow as a software product" perspective, I'd say we should have the default be `area|kind|priority|triage`, and then the configured value would replace the default (authoritative). That way, the default is sane and not specific to kubernetes, but then the configured option would be an override if the prow user wants to use "region" instead of "area" or what have you.

Reply:
:100: everything @cblecker said

Reply:
@cblecker :100: :tada: 

Made the plugin to be org/repo scoped + the behaviour that @cblecker mentioned. Ptal.